main/src/cgeo/geocaching/connector/gc/GCWebAPI.java

```diff
@@ -391,8 +396,16 @@ static boolean postLogTrackable(final String geocode, final String logDate, fina
     /**
      * https://www.geocaching.com/api/proxy/trackables?inCollection=false&skip=0&take=50
      */
-    static Single<TrackableInventoryEntry[]> getTrackableInventory() {
-        return getAPI("/trackables?inCollection=false&skip=0&take=50", TrackableInventoryEntry[].class);
+    static List<TrackableInventoryEntry> getTrackableInventory() {
+        final List<TrackableInventoryEntry> trackableInventoryEntries = new ArrayList<>();
+        int skip = 0;
+        TrackableInventoryEntry[] entries = null;
+        while (entries == null || entries.length == MAX_TAKE) {
```

Comment:
We can get rid of the first condition (and the `null`-initialization of `entries`) by using a do-while loop.

Reply:
Good point, thanks.